cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,249 @@ Cask Data Application Platform Release Notes
+
+- :cask-issue:`CDAP-12110` - Implemented Apache Commons JEXL based condition plugin which is available by default for the batch data pipelines.
+
+- :cask-issue:`CDAP-12111` - Added support for plugins of plugins. This allows the parent plugin to expose some APIs that its own plugins will implement and extend
+
+- :cask-issue:`CDAP-12114` - UI: Add ability to support custom label and custom icons for pipeline plugins
+
+- :cask-issue:`CDAP-12236` - Implements the design for Condition, a special type of pipeline plugin.
+
+- :cask-issue:`CDAP-10974` - BatchSource, BatchSink, BatchAggregator, BatchJoiner, and Transform plugins now have a way to get SettableArguments when preparing a run, which allows them to set arguments for the rest of the pipeline
+
+- :cask-issue:`CDAP-11618` - UI: Add Kafka Connection in Data Preparation
+
+
+
+Data Preparation Enhancements
+-----------------------------
+
+- :cask-issue:`CDAP-12092` - Adds Calculate directive to the directives dropdown in Data Prep.
+
+- :cask-issue:`CDAP-12118` - Adds Custom Transform directive to the directives dropdown in Data Prep.
+
+- :cask-issue:`CDAP-9530` - Adds point and click interaction in Dataprep UI to mask column data.
+
+- :cask-issue:`CDAP-9532` - Adds point and click interaction in Dataprep UI to encode/decode column data 
+
+- :cask-issue:`CDAP-11869` - Adds point and click interaction in Dataprep UI to parse Avro and Excel files.
+
+
+Other New Features
+------------------
+
+- :cask-issue:`CDAP-12232` - Added composite AND and OR trigger
+
+- :cask-issue:`CDAP-12382` - Added the capability of using plugin properties and runtime arguments from the triggering data pipeline as runtime arguments in the triggered data pipeline.
+
+- :cask-issue:`CDAP-12348` - Live migration of metrics tables from pre 4.3 tables to 4.3 salted metrics tables.
+
+- :cask-issue:`CDAP-10653` - Added a method to get pipeline arguments in the script plugins
+
+- :cask-issue:`CDAP-1952` - DatasetOutputCommiter's methods are now executed in the MapReduce ApplicationMaster, within OutputCommitter's commitJob/abortJob methods. The MapReduceContext.addOutput(Output.of(String, OutputFormatProvider)) API can no longer be used to add OutputFormatProviders that also implement the DatasetOutputCommitter interface. 
+
+- :cask-issue:`CDAP-11889` - Adds visualization for pipeline in UI. This helps visualizing runs, logs/warnings and data flowing through each node for each run in the pipeline.
+
+- :cask-issue:`CDAP-11912` - Added capabilities to trigger programs and data pipelines based on status of other programs and data pipelines.
+
+- :cask-issue:`CDAP-11940` - Adds the Create Microservice wizard to allow creating microservices in UI.
+
+- :cask-issue:`CDAP-11977` - UI: Add replace pattern column bulk action point and click
```

Comment:
Move to Data Preparation. also, ``Added point and click interaction for replacing column names in bulk``